### ISSUE TITLE
[css-grid-3][masonry] Enabling repeat(auto-fill, auto)

### ISSUE THREAD

**Issue by fantasai** [w3c/csswg-drafts#10915]
Google's earlier Alternate Masonry Proposal—which didn't allow for mixed track sizing and constrained all content-sized tracks to be the same size—allowed `repeat(auto-fill, auto)` as a track listing, computing `auto` as the largest [item size]/[span size].

This definition doesn't work with mixed track sizes, because the amount that an item contributes to each track can differ based on what it's spanning. Filing this issue to ask if we can still add this value, and if so, what does it mean? Does it also work in grid layout?

Note: There's a separate, but related, [issue about what the initial track listing should be](https://github.com/w3c/csswg-drafts/issues/10869), and whether it should be this.

**Comment by fantasai** [comment]
Copying over the key part of [Tab's comment](https://github.com/w3c/csswg-drafts/issues/10869#issuecomment-2356989394):

>  I've set down what I think is a reasonable definition (basically just restoring the old behavior from my draft) in https://drafts.csswg.org/css-grid-3/#masonry-intrinsic-repeat, along with an explanation of _why_ the definition is the way it is.

**Comment by fantasai** [comment]
See also earlier discussion in https://github.com/w3c/csswg-drafts/issues/9321

**Comment by tabatkins** [comment]
Yeah, my original draft's constraints meant that auto-sized repeats could be done *completely* correctly; there was no chance of spans covering different types of tracks. (You still had to ignore placement, of course, but it had less of an effect.)

Adapting that into this draft means that I can at best be heuristic. If you've only got span-1 items, it works correctly; if you do have all the tracks identical (which is what the initial value gives), it works correctly. It's only when you have a mixture of track sizes *and* spanning items that the heuristics have to kick in, and I think they still give a reasonable result.

**Comment by jensimmons** [comment]
We don't believe that `repeat(auto-fill, auto)` is a useful track definition. It's quite complex to explain why not, so please read our article about it, starting from [this link](https://webkit.org/blog/16026/css-masonry-syntax/#:~:text=Extending%20Masonry%20with%20repeat(auto%2Dareas%2C%20auto)).

**Comment by fantasai** [comment]
@jensimmons I think you're referring to the `auto-areas` value actually.

**Comment by yisibl** [comment]
Blink just implemented `repeat(auto-fill, auto)`.
https://chromium-review.googlesource.com/c/chromium/src/+/6692514

**Comment by alisonmaher** [comment]
> Blink just implemented repeat(auto-fill, auto).
https://chromium-review.googlesource.com/c/chromium/src/+/6692514

Yeah, this should now be available to try out behind the Masonry feature flag (chrome://flags/#css-masonry-layout) in Chrome Canary if you want to give it a try. I'm hoping that having an implementation to play with will help make it more tangible and aid in the conversations around whether this is a value we want, and if it is the right default track size for masonry.

**Comment by alisonmaher** [comment]
@fantasai this is the same issue as https://github.com/w3c/csswg-drafts/issues/9321 right? If so, which of these should we close out as a duplicate?
